server/src/document/DbtTextDocument.ts

```diff
@@ -90,6 +80,7 @@ export class DbtTextDocument {
     this.rawDocument = TextDocument.create(doc.uri, doc.languageId, doc.version, doc.text);
     this.compiledDocument = TextDocument.create(doc.uri, doc.languageId, doc.version, doc.text);
     this.diagnosticGenerator = new DiagnosticGenerator(this.dbtRepository);
+    this.completionProvider = new CompletionProvider(this.rawDocument, this.compiledDocument, this.dbtRepository, this.jinjaParser, destinationState);
```

Comment:
Now we create `DbtDefinitionProvider` in `server/src/lsp_server/LspServer.ts`. But `CompletionProvider` is created here. Can we generalize it somehow?